junit5-engine/src/main/java/org/junit/gen5/engine/junit5/descriptor/TestFactoryTestDescriptor.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.gen5.engine.junit5.descriptor;
+
+import static org.junit.gen5.commons.meta.API.Usage.Internal;
+import static org.junit.gen5.engine.junit5.execution.MethodInvocationContextFactory.methodInvocationContext;
+
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import org.junit.gen5.api.DynamicTest;
+import org.junit.gen5.api.extension.MethodInvocationContext;
+import org.junit.gen5.api.extension.TestExtensionContext;
+import org.junit.gen5.commons.JUnitException;
+import org.junit.gen5.commons.meta.API;
+import org.junit.gen5.engine.EngineExecutionListener;
+import org.junit.gen5.engine.TestExecutionResult;
+import org.junit.gen5.engine.UniqueId;
+import org.junit.gen5.engine.junit5.execution.JUnit5EngineExecutionContext;
+import org.junit.gen5.engine.junit5.execution.MethodInvoker;
+import org.junit.gen5.engine.junit5.execution.ThrowableCollector;
+import org.junit.gen5.engine.support.hierarchical.Leaf;
+import org.junit.gen5.engine.support.hierarchical.SingleTestExecutor;
+
+@API(Internal)
+public class TestFactoryTestDescriptor extends MethodTestDescriptor implements Leaf<JUnit5EngineExecutionContext> {
+
+	public static final String DYNAMIC_TEST_SEGMENT_TYPE = "dynamic-test";
+
+	public TestFactoryTestDescriptor(UniqueId uniqueId, Class<?> testClass, Method testMethod) {
+		super(uniqueId, testClass, testMethod);
+	}
+
+	@Override
+	public boolean isContainer() {
+		return true;
+	}
+
+	@Override
+	public boolean isTest() {
+		return true;
```

Comment:
- [ ] This should return `false`. However, then launcher will prune the test descriptor.